catkin_ws/src/vision/src/object_detection.py

```diff
@@ -79,85 +84,93 @@ def detect_on_image(raw_img, camera_id):
                     confidence.append(conf)
                     pred_obj_x, pred_obj_y, pred_obj_z = getObjectPositionFrontCam(bbox)
-                    obj_x.append(pred_obj_x)
-                    obj_y.append(pred_obj_y)
-                    obj_z.append(pred_obj_z) 
-                    obj_theta_z.append(None)
-                    extra_field.append(None)
+                    detectionFrame.x = pred_obj_x
+                    detectionFrame.y = pred_obj_y
+                    detectionFrame.z = pred_obj_z
+                    detectionFrame.theta_z = None
+                    detectionFrame.extra_field = None
+
+                    detectionFrameArray.append(detectionFrame)
                 elif global_class_name == "Gate": # GATE
-                    label.append(global_class_name)
+                    detectionFrame.label = global_class_name
                     confidence.append(conf)
                     theta_z = measureAngle(bbox)
                     pred_obj_x, pred_obj_y, pred_obj_z = getObjectPositionFrontCam(bbox)
-                    obj_x.append(pred_obj_x)
-                    obj_y.append(pred_obj_y)
-                    obj_z.append(pred_obj_z) 
-                    obj_theta_z.append(theta_z)
-                    extra_field.append(leftmost_gate_symbol) # 1 for earth, 0 for the other one
+                    detectionFrame.x = pred_obj_x
+                    detectionFrame.y = pred_obj_y
+                    detectionFrame.z = pred_obj_z
+                    detectionFrame.theta_z = theta_z
+                    detectionFrame.extra_field = leftmost_gate_symbol # 1 for earth, 0 for the other one
+
+                    detectionFrameArray.append(detectionFrame)
                 elif global_class_name == "Buoy": # BUOY
-                    label.append(global_class_name)
+                    detectionFrame.label = global_class_name
                     confidence.append(conf)
                     theta_z = measureAngle(bbox)
                     pred_obj_x, pred_obj_y, pred_obj_z = getObjectPositionFrontCam(bbox)
-                    obj_x.append(pred_obj_x)
-                    obj_y.append(pred_obj_y)
-                    obj_z.append(pred_obj_z) 
-                    obj_theta_z.append(theta_z)
-                    extra_field.append(None)
+                    detectionFrame.x = pred_obj_x
+                    detectionFrame.y = pred_obj_y
+                    detectionFrame.z = pred_obj_z
+                    detectionFrame.theta_z = theta_z
+                    detectionFrame.extra_field = None
+
+                    detectionFrameArray.append(detectionFrame)
 
                     for symbol_class_name, symbol_x, symbol_y, symbol_z, confidence in buoy_symbols:
-                        label.append(symbol_class_name)
-                        obj_x.append(symbol_x)
-                        obj_y.append(symbol_y)
-                        obj_z.append(symbol_z) 
+                        detectionFrame = VisionObject()
```

Comment:
this will overwrite the previously created instance of detectionFrame, but we want to add new detection Frames. So this should be renamed to something like symbolDetectionFrame

Reply:
I append the detection frame before entering the loop and I do it for each element in buoy_symbols

Reply:
oh my bad, ignore this then